### ISSUE TITLE
Ability to create Issue via Slack command

### ISSUE THREAD

**Issue by objectiveSee** [integrations/slack#379]
It would be very useful to be able to make a GitHub story directly from a Slack command. For example:

```
/github issue Fix the padding on the logout button
```

Thanks!

**Comment by philschatz** [comment]
👍 for an easier way to create Issues. I use an ad-hoc way to create new Issues... add a `:github:` reaction to any message and a [Project Card](https://github.com/blog/2272-introducing-projects-for-organizations) is created with a link back to the message.

The code (and screencap) are available [here](https://github.com/openstax/staxly/pull/18) but it would be nice to have something similar in an official app.

**Comment by wilhelmklopp** [comment]
We've played around with this idea a little bit a while back: https://youtu.be/JiR0mbxv280

Would be very curious to find out under what circumstances you typically want to create an issue right from Slack as opposed to going to GitHub to open it (where you will always have a more sophisticated interface then we'll be able to replicate in Slack with respect to markdown, context, milestones, labels, etc.)

**Comment by jphorn** [comment]
We'd REALLY REALLY REALLY love a way to (instantly) create GitHub issues from Slack updates, sort of like [Busybot does with their Zap emoji](https://support.busybot.com/article/52-creating-tasks-with-a-reaction-emoji) . But even just the ability to create issues directly from Slack would be welcome.

> Would be very curious to find out under what circumstances you typically want to create an issue right from Slack as opposed to going to GitHub to open it (where you will always have a more sophisticated interface then we'll be able to replicate in Slack with respect to markdown, context, milestones, labels, etc.)

Sometimes a non-dev team member reports an issue with the site in one of the channels. It would be most convenient to quickly send this to Github from Slack (via a reacji). Let devs do their formatting later in the Github interface ;-)

**Comment by james-lawrence** [comment]
@wilhelmklopp generally when someone else in the company talks about an issue, we'll want to create it directly into a github issue so it isn't lost.

**Comment by wilhelmklopp** [comment]
Thanks for the comments! That's super useful context.

I wonder if the following flow would make sense:
Reacji added to message -> User who added the reacji gets a DM from @github with a button -> Clicking that button will open up a Slack Dialog to create a new issue (pre-filled with the content of the message that was initially reacted to) -> Hitting submit on that dialog will actually open the issue in Slack.

_Also Note: Implementing this is currently blocked by our app not having the `reactions:read` scope. With workspace token apps that shouldn't be super hard to add, but the scope currently has a bit more access to data than we'd want_

**Comment by jphorn** [comment]
Hi @wilhelmklopp. That'd definitely work for us. That's more or like the same flow [Busybot uses](https://support.busybot.com/article/52-creating-tasks-with-a-reaction-emoji), although they use message buttons for the additional info.

![](http://take.ms/3L6HE)

**Comment by zackyap** [comment]
https://github.com/integrations/slack/issues/594

Since the release of the new Slack actions API, I think that's probably the easiest way to be able to both create an issue and add a certain conversation to an issue.

**Comment by jphorn** [comment]
Agreed. This would be best done through the new Actions API.

**Comment by wilhelmklopp** [comment]
There's a work in progress PR for it: #607

**Comment by wilhelmklopp** [comment]
Just updated this to say that you can create issues using `/github open owner/repo` 🎉 
Pls follow this issue for the corresponding action: https://github.com/integrations/slack/issues/594
Or the WIP PR in my previous comment

**Comment by danmash** [comment]
@wilhelmklopp is there a way to assign issue or output the link?

**Comment by wilhelmklopp** [comment]
@danmash hey! i'm not working on this project any more, but @philipbremer and @tebriel might be able to point you the right way 😄

**Comment by tebriel** [comment]
@danmash not right now, I'll add it to our feature request list though!

**Comment by danmash** [comment]
thanks!

**Comment by adamwarmoth-anduril** [comment]
Do you plan to support this with the Enterprise integration? Was very happy to see it's possible, then sad to realize it's not possible with Enterprise

**Comment by filipgolonka** [comment]
@adamwarmoth-anduril not built in into Integrations, but you can take a look into the library I've created: https://github.com/filipgolonka/slack-github-issue-creator

**Comment by garciasdos** [comment]
This would be very useful still in 2023 😄

**Comment by ashishjullia** [comment]
@wilhelmklopp is there a way to even pass the label name and other fields when creating/opening the issue via the slack command?
